applicationloadbalancer/v0.0.1/main.tf

```diff
@@ -0,0 +1,151 @@
+#Create an application load balancer. Can be either public or internal.
+resource "aws_lb" "loadbalancer" {
+  name                       = var.name
+  internal                   = var.load_balancer_internal
+  load_balancer_type         = "application"
+  security_groups            = [aws_security_group.allow_lb.id]
+  subnets                    = var.subnet_ids
+  drop_invalid_header_fields = true
+
+  enable_deletion_protection = var.enable_deletion_protection
+  idle_timeout               = 3600
+  access_logs {
+    bucket  = aws_s3_bucket.lb_logs.bucket
+    prefix  = "${var.name}-lb_logs"
+    enabled = true
+  }
+
+  tags = var.tags
+}
+
+#Create a HTTP listener that redirects all traffic to HTTPS
+resource "aws_lb_listener" "http" {
+  load_balancer_arn = aws_lb.loadbalancer.arn
+  port              = "80"
+  protocol          = "HTTP"
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+#Create a HTTPS listener that has a default response of 503
+resource "aws_lb_listener" "https" {
+  load_balancer_arn = aws_lb.loadbalancer.arn
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = var.ssl_policy
+  certificate_arn   = var.default_certificate_arn
+  default_action {
+    type = "fixed-response"
+    fixed_response {
+      content_type = "text/plain"
+      status_code  = "503"
+    }
+  }
+}
+
+#Attach any additional certificates to the HTTPS listener
+resource "aws_lb_listener_certificate" "certificate" {
+  count           = length(var.additional_certificate_arns)
+  listener_arn    = aws_lb_listener.https.arn
+  certificate_arn = var.additional_certificate_arns[count.index]
+}
+
+#Create the public security group
+data "aws_ec2_managed_prefix_list" "cloudfront" {
```

Comment:
This config will only allow use of this resource together with CloudFront. This should be optional, so the Ingress rules should be configurable